string output_dir
string output_prefix
string output_suffix
